gen/nvml/lib.go

```diff
@@ -99,10 +104,17 @@ func (l *library) load() error {
 // close the underlying library and ensure that the global pointer to the
 // library is set to nil to ensure that subsequent calls to open will reinitialize it.
 // Multiple calls to an already closed nvml library will return without error.
-func (l *library) close() error {
+func (l *library) close() (rerr error) {
 	l.Lock()
 	defer l.Unlock()
 
+	l.refcount.Dec()
+	defer l.refcount.IncOnError(rerr)
```

Comment:
What about when `refcount == 0` to start with and then does not get decreased to `-1` (because we're not allowing negative values). If there is then an error, our post-condition for calling `close` is then that `refcount == 1` and not `refcount == 0`.